packages/react/src/components/AppBar/TopNav.tsx

```diff
@@ -0,0 +1,281 @@
+  /**
+   * Application portal name.
+   */
+  portalName?: string;
+  /**
+   * URL for the App Nav logo.
+   */
+  settings?: MenuSettingsInterface[];
+  /**
+   * List of themes.
+   */
+  themes?: ThemeListInterface[];
+  /**
+   * Email of the logged user.
+   */
+  userEmail?: string;
+  /**
+   * Username of the logged user.
+   */
+  userName?: string;
+}
+
+export interface MenuSettingsInterface {
+  icon?: string | ReactElement;
+  name: string;
+  path: string;
+}
+
+export interface ThemeListInterface {
+  icon?: string | ReactElement;
+  name: string;
+}
+
+const COMPONENT_NAME: string = 'TopNav';
+
+const TopNav: FC<TopNavProps> & WithWrapperProps = (props: TopNavProps): ReactElement => {
+  const {
+    className,
+    isLeftNavActive,
+    logoUrl,
+    avatarUrl,
+    settings,
+    themes,
+    mobileLogoUrl,
+    portalName,
+    userName,
+    userEmail,
+    homePath,
+    handleLogOut,
+    handleLeftNav,
+    ...rest
+  } = props;
+
+  const [anchorElUser, setAnchorElUser] = React.useState<null | HTMLElement>(null);
+  const [selectedTheme, setSelectedTheme] = useState('System Default');
+
+  const classes: string = clsx('oxygen-top-nav', className);
+
+  const open: boolean = Boolean(anchorElUser);
+
+  const handleOpenUserMenu = (event: React.MouseEvent<HTMLElement>): void => {
+    setAnchorElUser(event.currentTarget);
+  };
+
+  const handleCloseUserMenu = (): void => {
+    setAnchorElUser(null);
+  };
+
+  const defaultThemes: ThemeListInterface[] = [
+    {icon: <DisabledByDefaultOutlinedIcon />, name: 'System Default'},
+    {icon: <LightModeOutlinedIcon />, name: 'Light'},
+    {icon: <DarkModeOutlinedIcon />, name: 'Dark'},
+    {icon: <ContrastOutlinedIcon />, name: 'High Contrast'},
+  ];
+
+  const onThemeChange = (theme: string) => () => {
+    setSelectedTheme(theme);
+  };
+
+  const isMobile: boolean = useMediaQuery('(max-width:768px)');
+
+  return (
+    <MuiAppBar color="default" position="static" className={classes} {...rest}>
+      <Container maxWidth="xl">
+        <Toolbar disableGutters>
+          {isLeftNavActive && (
+            <IconButton aria-label="Menu Icon" onClick={handleLeftNav} className="menu-icon-button">
+              <MenuIcon />
+            </IconButton>
+          )}
+          <Box
+            component={homePath ? Link : Box}
+            className={homePath ? 'logo-box with-link' : 'logo-box'}
+            href={homePath}
+            color="inherit"
+            underline="none"
+            aria-label="Home Page"
+          >
+            {(logoUrl || mobileLogoUrl) && (
+              <Image
+                className="logo"
+                src={isMobile && mobileLogoUrl ? mobileLogoUrl : logoUrl}
+                alt="Logo"
+                width="auto"
+              />
+            )}
+            <Typography variant="h6" className="portal-name">
+              {portalName}
+            </Typography>
+          </Box>
+
+          <Box className="settings">
```

Comment:
If `settings` is falsy, the `Box` component will be rendered with this implementation. Is that expected?

Reply:
The positioning of the user dropdown in set by the setting's box's flex-grow property (`flex-grow: 1`). Can fix the user dropdown to the right side of the top nav as well. WDYT?

Reply:
resolved with latest commit